pygatt/backends/bluez/device.py

```diff
@@ -0,0 +1,222 @@
+                                      service=o.Service, uuid=uuid))
+            el_gatt_o = o[self._dbus.GATT_CHAR_INTERFACE]
+            self._subscribed_characteristics[uuid] = set((callback,))
+            el_gatt_o.StartNotify()
+
+    def unsubscribe(self, uuid):
+        if uuid not in self._subscribed_characteristics:
+            return
+
+        objs = self._dbus.objects_by_property({'UUID': uuid},
+                interface=self._dbus.GATT_CHAR_INTERFACE)
+        for o in objs:
+            el_gatt_o = o[self._dbus.GATT_CHAR_INTERFACE]
+            if el_gatt_o.Notifying:
+                el_gatt_o.StopNotify()
+            del(self._subscribed_characteristics[uuid])
+
+    def properties_changed(self, interface, changed, invalidated,
+                           service=None, uuid=None):
+        log.debug("Property changed on service: %s, uuid %s", service, uuid)
+        if uuid is not None and uuid in self._subscribed_characteristics:
+            for cb in self._subscribed_characteristics[uuid]:
+                cb(interface, changed, invalidated, service=service, uuid=uuid)
+        elif uuid is not None:
+            log.error("No subscription for UUID {}".format(uuid))
+
+    def _notification_handles(self, uuid):
+        uuid_parts = uuid.split('-')
+        uuid_parts[0] = ('%08d' % (int(uuid_parts[0]) + 1))
+        uuid_notification = '-'.join(uuid_parts)
+        return uuid, uuid_notification
+
+    def get_handle(self, char_uuid):
+        log.warning('get_handle not implemented - returning given uuid')
+        return char_uuid
+
+    @connection_required
+    def bond(self, *args, **kwargs):
+        raise NotImplementedError()
+
+    @connection_required
+    def clear_bond(self, address=None):
+        raise NotImplementedError()
+
+    @connection_required
+    def char_read(self, uuid, *args, **kwargs):
+        """
+        Reads a Characteristic by uuid.
+        :param bledevice: BluezBLEDevice instance
+        :param uuid: UUID of Characteristic to read.
+        :type uuid: str
+        :return: bytearray of result.
+        :rtype: bytearray
+        """
+        log.debug("Char read from %s", uuid)
+        objects = self._dbus.get_managed_objects()
+        for path, ifaces in objects.iteritems():
+            iface = ifaces.get(self._dbus.GATT_CHAR_INTERFACE)
+            if iface is None or iface['UUID'] != uuid:
+                if iface is not None: print(iface['UUID'])
+                continue
+            dbus_obj = self._dbus.object_by_path(path,
+                    interface=self._dbus.GATT_CHAR_INTERFACE)
+            v = dbus_obj.ReadValue({})
+            return bytearray(v)
+        raise Exception("UUID {} not found".format(uuid))
+
+    @connection_required
+    def char_write(self, uuid, value, wait_for_response=False):
+        objs = self._dbus.objects_by_property({'UUID': uuid},
+                                                      base_path='/')
```

Comment:
The `objects_by_property` method is missing an interface argument here: `interface=self._dbus.GATT_CHAR_INTERFACE`.